isthmus/src/test/java/io/substrait/isthmus/SimpleExtendedExpressionsTest.java

```diff
@@ -0,0 +1,13 @@
+package io.substrait.isthmus;
+
+import java.io.IOException;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.junit.jupiter.api.Test;
+
+public class SimpleExtendedExpressionsTest extends ExtendedExpressionTestBase {
+
+  @Test
+  public void filter() throws IOException, SqlParseException {
+    assertProtoExtendedExpressionRoundrip("N_NATIONKEY > 18");
```

Comment:
I'm curious about this. Does the Calcite let you parse expressions by themselves outside the context of a query?

Reply:
This test `assertProtoExtendedExpressionRoundrip ` internally pass "the expression + tpchSchemaCreateStatements"

Then, there is a documented parse expression for example for [expr = "1 in (1,2,3)"](https://github.com/apache/calcite/blob/dc3f11bf9b8c7814eaaf7cbbf96a24b0c2b50456/core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java#L4932-L4935).

It is also possible to parse expression passing internal definition using  [Map<String, RexNode> nameToNodeMap](https://github.com/apache/calcite/blob/c6031ca15579b88bd392a56f4588fdd9a29d1071/core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java#L2153-L2155), ... my current approach is that I [just created that manually](https://github.com/substrait-io/substrait-java/blob/333989ffa70efc18c6707c3cf670bac826925fe1/isthmus/src/main/java/io/substrait/isthmus/SqlToSubstrait.java#L286-L301), my next step is to figure out how to populate/create `Map<String, RexNode> nameToNodeMap]` dynamically using schema definition provided.

```
  /**
   * Converts an expression from {@link SqlNode} to {@link RexNode} format,
   * mapping identifier references to predefined expressions.
   *
   * @param node          Expression to translate
   * @param nameToNodeMap map from String to {@link RexNode}; when an
   *                      {@link SqlIdentifier} is encountered, it is used as a
   *                      key and translated to the corresponding value from
   *                      this map
   * @return Converted expression
   */
  public RexNode convertExpression(
      SqlNode node,
      Map<String, RexNode> nameToNodeMap) {
```